roachtest: lib/pq failed

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.lib/pq [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/18906414?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/18906414?buildTab=artifacts#/lib/pq) on master @ [fa8342d50c1bdfbc307f4bca9bc0d34da4475258](https://github.com/cockroachdb/cockroach/commits/fa8342d50c1bdfbc307f4bca9bc0d34da4475258):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.2.0-alpha.00000000-dev-fa8342d50c1bdfbc307f4bca9bc0d34da4475258
Tests run against lib/pq v1.10.5
216 Total Tests Run
189 tests passed
27 tests failed
7 tests skipped
0 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: pq.TestCopyInWrongType - unknown (unexpected)
--- FAIL: pq.TestCopyInMultipleValues - unknown (unexpected)
--- FAIL: pq.TestCopyFromError - unknown (unexpected)
For a full summary look at the lib/pq artifacts 
An updated blocklist (libPQBlocklist) is available in the artifacts' lib/pq log
(cluster.go:2490).Run: context canceled
test artifacts and logs in: /artifacts/lib/pq/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*lib/pq.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48059

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.lib/pq [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18906403?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18906403?buildTab=artifacts#/lib/pq) on master @ [fa8342d50c1bdfbc307f4bca9bc0d34da4475258](https://github.com/cockroachdb/cockroach/commits/fa8342d50c1bdfbc307f4bca9bc0d34da4475258):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.2.0-alpha.00000000-dev-fa8342d50c1bdfbc307f4bca9bc0d34da4475258
Tests run against lib/pq v1.10.5
216 Total Tests Run
189 tests passed
27 tests failed
7 tests skipped
0 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: pq.TestCopyInWrongType - unknown (unexpected)
--- FAIL: pq.TestCopyInMultipleValues - unknown (unexpected)
--- FAIL: pq.TestCopyFromError - unknown (unexpected)
For a full summary look at the lib/pq artifacts 
An updated blocklist (libPQBlocklist) is available in the artifacts' lib/pq log
(cluster.go:2490).Run: context canceled
test artifacts and logs in: /artifacts/lib/pq/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18906403/lib-pq/1740300097248/1740300286668)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*lib/pq.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.lib/pq [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/18912543?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/18912543?buildTab=artifacts#/lib/pq) on master @ [cbb292c487ea7413772a5c3c349f93467264c190](https://github.com/cockroachdb/cockroach/commits/cbb292c487ea7413772a5c3c349f93467264c190):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.2.0-alpha.00000000-dev-cbb292c487ea7413772a5c3c349f93467264c190
Tests run against lib/pq v1.10.5
216 Total Tests Run
189 tests passed
27 tests failed
7 tests skipped
0 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: pq.TestCopyInWrongType - unknown (unexpected)
--- FAIL: pq.TestCopyFromError - unknown (unexpected)
--- FAIL: pq.TestCopyInMultipleValues - unknown (unexpected)
For a full summary look at the lib/pq artifacts 
An updated blocklist (libPQBlocklist) is available in the artifacts' lib/pq log
(cluster.go:2490).Run: context canceled
test artifacts and logs in: /artifacts/lib/pq/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=true</code>
 - <code>metamorphicLeases=epoch</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*lib/pq.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.lib/pq [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18912532?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18912532?buildTab=artifacts#/lib/pq) on master @ [cbb292c487ea7413772a5c3c349f93467264c190](https://github.com/cockroachdb/cockroach/commits/cbb292c487ea7413772a5c3c349f93467264c190):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.2.0-alpha.00000000-dev-cbb292c487ea7413772a5c3c349f93467264c190
Tests run against lib/pq v1.10.5
216 Total Tests Run
189 tests passed
27 tests failed
7 tests skipped
0 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: pq.TestCopyInWrongType - unknown (unexpected)
--- FAIL: pq.TestCopyFromError - unknown (unexpected)
--- FAIL: pq.TestCopyInMultipleValues - unknown (unexpected)
For a full summary look at the lib/pq artifacts 
An updated blocklist (libPQBlocklist) is available in the artifacts' lib/pq log
(cluster.go:2490).Run: context canceled
test artifacts and logs in: /artifacts/lib/pq/cpu_arch=arm64/run_1
```

Parameters:
 - <code>arch=arm64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=false</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18912532/lib-pq/1740387937501/1740388143072)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*lib/pq.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.lib/pq [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/18923269?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/18923269?buildTab=artifacts#/lib/pq) on master @ [8bd38972db32f5e683ba6dc0eb599334919919e5](https://github.com/cockroachdb/cockroach/commits/8bd38972db32f5e683ba6dc0eb599334919919e5):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.2.0-alpha.00000000-dev-8bd38972db32f5e683ba6dc0eb599334919919e5
Tests run against lib/pq v1.10.5
216 Total Tests Run
189 tests passed
27 tests failed
7 tests skipped
0 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: pq.TestCopyFromError - unknown (unexpected)
--- FAIL: pq.TestCopyInMultipleValues - unknown (unexpected)
--- FAIL: pq.TestCopyInWrongType - unknown (unexpected)
For a full summary look at the lib/pq artifacts 
An updated blocklist (libPQBlocklist) is available in the artifacts' lib/pq log
(cluster.go:2491).Run: context canceled
test artifacts and logs in: /artifacts/lib/pq/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=false</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*lib/pq.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.lib/pq [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18923258?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18923258?buildTab=artifacts#/lib/pq) on master @ [8bd38972db32f5e683ba6dc0eb599334919919e5](https://github.com/cockroachdb/cockroach/commits/8bd38972db32f5e683ba6dc0eb599334919919e5):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.2.0-alpha.00000000-dev-8bd38972db32f5e683ba6dc0eb599334919919e5
Tests run against lib/pq v1.10.5
216 Total Tests Run
189 tests passed
27 tests failed
7 tests skipped
0 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: pq.TestCopyFromError - unknown (unexpected)
--- FAIL: pq.TestCopyInMultipleValues - unknown (unexpected)
--- FAIL: pq.TestCopyInWrongType - unknown (unexpected)
For a full summary look at the lib/pq artifacts 
An updated blocklist (libPQBlocklist) is available in the artifacts' lib/pq log
(cluster.go:2491).Run: context canceled
test artifacts and logs in: /artifacts/lib/pq/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=false</code>
 - <code>metamorphicLeases=epoch</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18923258/lib-pq/1740470097130/1740470265980)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*lib/pq.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.lib/pq [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/18934383?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/18934383?buildTab=artifacts#/lib/pq) on master @ [c44b3134d23d6037e1d5d5e12461a6914aa588e3](https://github.com/cockroachdb/cockroach/commits/c44b3134d23d6037e1d5d5e12461a6914aa588e3):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v25.2.0-alpha.00000000-dev-c44b3134d23d6037e1d5d5e12461a6914aa588e3
Tests run against lib/pq v1.10.5
216 Total Tests Run
189 tests passed
27 tests failed
7 tests skipped
0 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: pq.TestCopyInWrongType - unknown (unexpected)
--- FAIL: pq.TestCopyFromError - unknown (unexpected)
--- FAIL: pq.TestCopyInMultipleValues - unknown (unexpected)
For a full summary look at the lib/pq artifacts 
An updated blocklist (libPQBlocklist) is available in the artifacts' lib/pq log
(cluster.go:2491).Run: context canceled
test artifacts and logs in: /artifacts/lib/pq/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>metamorphicBufferedSender=false</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*lib/pq.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/cmd/roachtest/tests/libpq_blocklist.go
+++ b/pkg/cmd/roachtest/tests/libpq_blocklist.go
@@ -30,6 +30,11 @@ var libPQBlocklist = blocklist{
 	"pq.TestRowsColumnTypes":                         "41688",
 	"pq.TestRuntimeParameters":                       "12137",
 	"pq.TestStringWithNul":                           "26366",
+	// The following tests fail because they weren't designed for the
+	// autocommit_before_ddl behaviour in CRDB.
+	"pq.TestCopyInWrongType":      "142038",
+	"pq.TestCopyInMultipleValues": "142038",
+	"pq.TestCopyFromError":        "142038",
 }
 
 // The test names here do not include "pq." since `go test -list` returns

